### ISSUE TITLE
QUESTION: Is it possible to configure FILESTREAM Feature via xSQLServer 

### ISSUE THREAD

**Issue by dvdgutzmann** [PowerShell/xSQLServer#97]
As far as my googling goes it is possible to enable FILESTREAM feature in two ways:

- [ Install SQL Server 2016 from the Command Prompt/Config file/ SysPrep/ etc...](https://msdn.microsoft.com/en-us/library/ms144259.aspx)
- [Enable and Configure FILESTREAM by using SQL Server Configuration Manager](https://technet.microsoft.com/en-us/library/cc645923(v=sql.130).aspx)

Is it also possible to enable FILESTREAM via DSC? If not, what would be the proper way to submit a feature request? 

Thank you!

**Comment by aultt** [comment]
Yes take a look at xSQLServerConfiguration.  Specifically look at this example --> https://github.com/PowerShell/xSQLServer/blob/dev/Examples/xSQLServerConfiguration.ps1 
This gives you the ability to configure anything that is in sp_configure.

**Comment by dvdgutzmann** [comment]
Thank you for the quick response @aultt! I'm not shure though, if reconfiguring _filestream_access_level_ would be enough to enable this _feature_.

[Enable and Configure FILESTREAM](https://technet.microsoft.com/en-us/library/cc645923(v=sql.130).aspx) states
>  Before you can start to use FILESTREAM, you must enable FILESTREAM on the instance of the SQL Server Database Engine.

In the meantime I found people doing it via WMI ([C#](https://stackoverflow.com/questions/2942469/is-there-an-alternative-of-using-a-vbscript-for-enabling-filestream-after-sql-ser) [VB](https://sqlsrvengine.codeplex.com/wikipage?title=FileStreamEnable&referringTitle=Home&ProjectName=sqlsrvengine)). I would like to avoid going down that particular road.

I wonder if there is a possibility to achive enabling FILESTREAM on the instance solely by means of DSC. If not, wouldn't that be something for xSQLServerSetup ?

Thank you for your time!

**Comment by dvdgutzmann** [comment]
I realise just now, that the title of the issue might have been misleading. My question is about **enabling** filestream, not only about  configuring it. Thank you.

**Comment by aultt** [comment]
Sorry you are correct I was looking at configuring Filestream.  To enable it there are a couple options as you pointed out.  One at installation time which would require modification to xSQLServerSetup.  Currently it doesn't have this capability. Since this can be done through WMI it could also be done via a DSC Script task.  If you are looking to build new servers with this feature enabled I would suggest entering an issue requesting for the ability to be added to xSQLServerSetup

**Comment by quillypowers** [comment]
JIC someone stumbles on this, below will enable
Registry RegistrySqlEnableFilestream
		{
			Ensure      = "Present"  # You can also set Ensure to "Absent"
			Key         = "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SQL Server\MSSQL13.MSSQLSERVER\MSSQLServer\Filestream"
			ValueType   = "Dword"
			ValueName   = "EnableLevel"
			ValueData   = "3"
		}

But this doesn't work to configure.  Probably because that's not passed in as a string 
    	SqlConfiguration SQLFilestreamAccess
        {

            InstanceName   = 'MSSQLSERVER'
            OptionName     = 'filestream_access_level'
            OptionValue    = 2
            RestartService = $false
			DependsOn  = '[SqlSetup]DefaultInstance'
        }
